### PR TITLE
Documentation refactoring towards 2.0

### LINKED ISSUE #132: Shutdown the project Wiki
Wiki pages on GitHub contain the outdated information, which is heavily duplicated with README. The documentation should be merged into the in-REPO docs, and then Wiki can be disabled

### PR DESCRIPTION
**Pull request by oleg-nenashev**
- [x] Decouple the documentation to several pages
- [x] Cleanup wording 
- [x] Add changelog
- [x] Ensure all internal and external links are correct
- [x] Integrate Wiki to close #132
- [x] Add some missing basic information about the installation flow

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
@reviewbybees

**Comment by oleg-nenashev** [comment]
Ideally it requires a full re-read of the new text

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+Release Notes
+====
+
+Below you can release notes for the trunk version of WinSW.
```
Comment (jtnord) [inline CHANGELOG.md]: can ... find or read?

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+##### 2.0
+
+Release date: Coming Soon
+
+Improvements:
+* Introduce a concept of WinSW extensions, which allow extending the wrapper's behavior
```
Comment (jtnord) [inline CHANGELOG.md]: Introduce**d** ...  which allow**s**

File: CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+([PR #42](https://github.com/kohsuke/winsw/pull/42)).
+ * Right now the extensions are not guaranteed to be binary compatible
```
Comment (jtnord) [inline CHANGELOG.md]: just "The extension binary backward comparability is not guaranteed"

File: CHANGELOG.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  * Engine does not support their inclusion from external DLLs (but you can repackage the executable if you want to extend WinSW)
```
Comment (jtnord) [inline CHANGELOG.md]: The engine does not support extensions in external DLLs

File: CHANGELOG.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+* `SharedDirectoriesMapper` extension for mapping shared directories when the Windows service starts up 
+([PR #42](https://github.com/kohsuke/winsw/pull/42)).
+* Migrate event logging handlers to log4j
```
Comment (jtnord) [inline CHANGELOG.md]: Migrated event logging to log4j

File: CHANGELOG.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+##### 2.0
+
+Release date: Coming Soon
+
+Improvements:
+* Introduce a concept of WinSW extensions, which allow extending the wrapper's behavior
+([PR #42](https://github.com/kohsuke/winsw/pull/42)).
+ * Right now the extensions are not guaranteed to be binary compatible
+  * Engine does not support their inclusion from external DLLs (but you can repackage the executable if you want to extend WinSW)
+* `SharedDirectoriesMapper` extension for mapping shared directories when the Windows service starts up
```
Comment (jtnord) [inline CHANGELOG.md]: `SharedDirectoriesMapper` extension for mapping shared directories when the Windows service starts

File: CHANGELOG.md [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+Fixed issues:
+
+* Fix the version number in the executable file metadata and CLI
+
+##### 1.19
+
+Release date: Aug 02, 2016 
+
+No functional changes.
```
Comment (jtnord) [inline CHANGELOG.md]: no bullet?

File: doc/installation.md [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+
+Once the start button is clicked, Windows will start `myapp.exe`, 
+  then `myapp.exe` will launch the executable specified in the configuration file (Java in this case). 
+  If this process dies, `myapp.exe` will exit itself, and the service will be considered stopped.
+  
+### Extra configuration options
+  
+#### Making WinSW compatible with .NET runtime 4.0+
+
+Modern versions of Windows (e.g. Windows Server 2012 or Windows 10) do not ship with .NET runtime `2.0`, which is what `winsw.exe` is built against.
```
Comment (jtnord) [inline doc/installation.md]: I thought this was the default?  If not why don't we just embed this info in the binary?

File: CHANGELOG.md (hunk 1 above)
Comment (oleg-nenashev) [inline CHANGELOG.md]: yep

File: doc/installation.md (hunk 8 above)
Comment (oleg-nenashev) [inline doc/installation.md]: We should, of course. In this doc I've just documented the current state

**Comment by oleg-nenashev** [comment]
> A couple of typos or suggestions for changes and a clarification about the .net framework version.

Will fix typos. Regarding the .NET version, it will be fixed in a separate PR for #102 . This is just a first refactoring step, which should allow parallelizing other changes

**Comment by oleg-nenashev** [comment]
Merged since I have a bee from @jtnord , and only typos have been fixed

**Pull request merged**